### ISSUE TITLE
Crash rendering window when parsing GraphQL enum value

### ISSUE THREAD

**Issue by 0x524c** [getinsomnia/insomnia#5470]
### Expected Behavior

// Stack error
```sh
GraphQLError: Value "" does not exist in "OauthProviders" enum.
    at tc.parseValue (file:///Applications/Insomnia.app/Contents/Resources/app.asar/index.70753967.js:507:10797)
    at ev (file:///Applications/Insomnia.app/Contents/Resources/app.asar/index.70753967.js:651:20522)
    at ev (file:///Applications/Insomnia.app/Contents/Resources/app.asar/index.70753967.js:651:19335)
    at file:///Applications/Insomnia.app/Contents/Resources/app.asar/index.70753967.js:651:19879
    at Array.map (<anonymous>)
    at M$ (file:///Applications/Insomnia.app/Contents/Resources/app.asar/index.70753967.js:651:20809)
    at ev (file:///Applications/Insomnia.app/Contents/Resources/app.asar/index.70753967.js:651:19659)
    at ev (file:///Applications/Insomnia.app/Contents/Resources/app.asar/index.70753967.js:651:19335)
    at file:///Applications/Insomnia.app/Contents/Resources/app.asar/index.70753967.js:651:19019
    at Array.forEach (<anonymous>)
    at XNe (file:///Applications/Insomnia.app/Contents/Resources/app.asar/index.70753967.js:651:18928)
    at file:///Applications/Insomnia.app/Contents/Resources/app.asar/index.70753967.js:651:18866
    at x (file:///Applications/Insomnia.app/Contents/Resources/app.asar/index.70753967.js:436:14257)
    at si.<anonymous> (file:///Applications/Insomnia.app/Contents/Resources/app.asar/index.70753967.js:436:16213)
    at file:///Applications/Insomnia.app/Contents/Resources/app.asar/index.70753967.js:412:10365
    at si.setOption (file:///Applications/Insomnia.app/Contents/Resources/app.asar/index.70753967.js:417:1982)
    at file:///Applications/Insomnia.app/Contents/Resources/app.asar/index.70753967.js:676:5014
    at AE (file:///Applications/Insomnia.app/Contents/Resources/app.asar/index.70753967.js:67:24458)
    at lh (file:///Applications/Insomnia.app/Contents/Resources/app.asar/index.70753967.js:67:42799)
    at file:///Applications/Insomnia.app/Contents/Resources/app.asar/index.70753967.js:67:41058
    at N (file:///Applications/Insomnia.app/Contents/Resources/app.asar/index.70753967.js:52:1647)
    at Immediate.re [as _onImmediate] (file:///Applications/Insomnia.app/Contents/Resources/app.asar/index.70753967.js:52:2021)
    at process.processImmediate (node:internal/timers:466:21)
```

// exported data collection that manifest the crash
```JSON
{
   "_type":"export",
   "__export_format":4,
   "__export_date":"2022-11-30T01:24:18.241Z",
   "__export_source":"insomnia.desktop.app:v2022.7.0-beta.6",
   "resources":[
      {
         "_id":"req_d5c396fa80e241c38c1f0546f6fbe60c",
         "parentId":"fld_134c6bdb51f54e39ab2711599799aa5b",
         "modified":1669771213545,
         "created":1669760659023,
         "url":"{{ _.server }}",
         "name":"FederatedSignup",
         "description":"",
         "method":"POST",
         "body":{
            "mimeType":"application/graphql",
            "text":"{\"query\":\"mutation FederatedSignup($UserInput: UserInput!, $FUserInput: FUserInput!) {\\n  FederatedSignup(UserInput: $UserInput, FUserInput: $FUserInput)   {\\n     __typename\\n    ... on MutationFederatedSignupSuccess {\\n       data {\\n         user {\\n           id\\n           pk\\n           provider\\n           subject\\n           userId\\n        }\\n      }\\n    }\\n  }\\n}\\n\",\"variables\":{\"UserInput\":{\"name\":\"Rogerio\",\"email\":\"user25@gmail.com\",\"password\":\"123456\"},\"FUserInput\":{\"userId\":\"108170005730914745289\",\"subject\":\"user25@gmail.com\",\"provider\":\"\"}},\"operationName\":\"Signup\"}"
         },
         "parameters":[
            
         ],
         "headers":[
            {
               "name":"Content-Type",
               "value":"application/json",
               "id":"pair_7ca4f5cd0b034ffaba3f8c17a09cb266"
            },
            {
               "name":"Authorization",
               "value":"Bearer {% variable 'APP_TOKEN' %}",
               "id":"pair_53357334a8e64014a545638adef887b7",
               "disabled":false
            },
            {
               "name":"Authorization",
               "value":"Bearer {% response 'body', 'req_fd8f601b4f7844e9bdae99aed930e52a', 'b64::JC5kYXRhLnNpZ25pbi5kYXRhLnRva2Vu::46b', 'never', 60 %}",
               "id":"pair_c118d752a77d4488aaf87c06a9aab95c",
               "disabled":true
            }
         ],
         "authentication":{
            
         },
         "metaSortKey":-1565089896241.25,
         "isPrivate":false,
         "settingStoreCookies":true,
         "settingSendCookies":true,
         "settingDisableRenderRequestBody":false,
         "settingEncodeUrl":true,
         "settingRebuildPath":true,
         "settingFollowRedirects":"global",
         "_type":"request"
      },
      {
         "_id":"fld_134c6bdb51f54e39ab2711599799aa5b",
         "parentId":"wrk_da8ad0bc8d3a46a7b084622b753f78bf",
         "modified":1669429258339,
         "created":1669429258339,
         "name":"mutate",
         "description":"",
         "environment":{
            
         },
         "environmentPropertyOrder":null,
         "metaSortKey":-1669429258339,
         "_type":"request_group"
      },
      {
         "_id":"wrk_da8ad0bc8d3a46a7b084622b753f78bf",
         "parentId":null,
         "modified":1660350418135,
         "created":1655745342238,
         "name":"mkt",
         "description":"",
         "scope":"collection",
         "_type":"workspace"
      },
      {
         "_id":"env_5fc09f74bb6141fd80da08deb2c302c9",
         "parentId":"wrk_da8ad0bc8d3a46a7b084622b753f78bf",
         "modified":1655745342264,
         "created":1655745342264,
         "name":"Base Environment",
         "data":{
            "server":"http://localhost:4000/graphql",
            "AdminId":"c83fc436-aff2-49e1-a4b4-669f641fbd6b",
            "CintiaId":"95c19e32-be9a-415a-a496-0705c181e245",
            "RandomId":"2a76ad03-72bd-4d05-9af2-945622b834ed"
         },
         "dataPropertyOrder":{
            "&":[
               "server",
               "AdminId",
               "CintiaId",
               "RandomId"
            ]
         },
         "color":null,
         "isPrivate":false,
         "metaSortKey":1655315179058,
         "_type":"environment"
      },
      {
         "_id":"jar_8f8306faa14249ca9c8778f99fe85370",
         "parentId":"wrk_da8ad0bc8d3a46a7b084622b753f78bf",
         "modified":1668101457730,
         "created":1655745342277,
         "name":"Default Jar",
         "cookies":[
            
         ],
         "_type":"cookie_jar"
      },
      {
         "_id":"spc_1c387b7ce5b542bda3cd300977010452",
         "parentId":"wrk_da8ad0bc8d3a46a7b084622b753f78bf",
         "modified":1655745342340,
         "created":1655745342280,
         "fileName":"wavechat",
         "contents":"",
         "contentType":"yaml",
         "_type":"api_spec"
      },
      {
         "_id":"env_4c1effe7124449d28ab013d2a7508a05",
         "parentId":"env_5fc09f74bb6141fd80da08deb2c302c9",
         "modified":1669422980924,
         "created":1655745342266,
         "name":"localhost",
         "data":{
            "server":"http://localhost:4000/graphql",
            "AdminId":"0622fe40-ebc8-483b-b0b9-7146caf9deb1",
            "CintiaId":"c219fd65-0995-4fd0-9c66-d78bc91fa4d8",
            "RandomId":"70cb1940-d634-418b-ac98-4cca693927c5",
            "AdminPass":"admin123",
            "UserPass":"123456",
            "loginCintia":{
               "email":"cintia.uff@gmail.com",
               "password":"123456"
            },
            "loginRoger":{
               "email":"rogeriolucas@gmail.com",
               "password":"admin123"
            },
            "APP_TOKEN":"eyJhbGciOiJFZERTQSIsInR5cCI6IkpXSyJ9.eyJwZXJtaXNzaW9ucyI6WzIsMyw0LDVdLCJpYXQiOjE2Njk0MjI5NDcsIm5iZiI6MTY2OTQyMjk0NywiZXhwIjoxNzAwOTU4OTQ3fQ.G-i_Jd19PnuTIzV-vVi0Egn699_HpMCUTFECcecTJUIIdK4jCBQkhKbIdba-EzCheyxF2Lf24XGvCcU6UVWaCA"
         },
         "dataPropertyOrder":{
            "&":[
               "server",
               "AdminId",
               "CintiaId",
               "RandomId",
               "AdminPass",
               "UserPass",
               "loginCintia",
               "loginRoger",
               "APP_TOKEN"
            ],
            "&~|loginCintia":[
               "email",
               "password"
            ],
            "&~|loginRoger":[
               "email",
               "password"
            ]
         },
         "color":"#51e34f",
         "isPrivate":false,
         "metaSortKey":1655508376577,
         "_type":"environment"
      },
      {
         "_id":"env_0a3c20a0398f4c8491d4a65d875f354f",
         "parentId":"env_5fc09f74bb6141fd80da08deb2c302c9",
         "modified":1669422985626,
         "created":1660828934080,
         "name":"rpi4-1",
         "data":{
            "server":"http://192.168.1.4:4000/graphql",
            "AdminId":"0622fe40-ebc8-483b-b0b9-7146caf9deb1",
            "CintiaId":"c219fd65-0995-4fd0-9c66-d78bc91fa4d8",
            "RandomId":"70cb1940-d634-418b-ac98-4cca693927c5",
            "AdminPass":"admin123",
            "UserPass":"123456",
            "loginCintia":{
               "email":"cintia.uff@gmail.com",
               "password":"123456"
            },
            "loginRoger":{
               "email":"rogeriolucas@gmail.com",
               "password":"admin123"
            },
            "APP_TOKEN":"eyJhbGciOiJFZERTQSIsInR5cCI6IkpXSyJ9.eyJwZXJtaXNzaW9ucyI6WzIsMyw0LDVdLCJpYXQiOjE2Njk0MjI5NDcsIm5iZiI6MTY2OTQyMjk0NywiZXhwIjoxNzAwOTU4OTQ3fQ.G-i_Jd19PnuTIzV-vVi0Egn699_HpMCUTFECcecTJUIIdK4jCBQkhKbIdba-EzCheyxF2Lf24XGvCcU6UVWaCA"
         },
         "dataPropertyOrder":{
            "&":[
               "server",
               "AdminId",
               "CintiaId",
               "RandomId",
               "AdminPass",
               "UserPass",
               "loginCintia",
               "loginRoger",
               "APP_TOKEN"
            ],
            "&~|loginCintia":[
               "email",
               "password"
            ],
            "&~|loginRoger":[
               "email",
               "password"
            ]
         },
         "color":"#51e34f",
         "isPrivate":false,
         "metaSortKey":1657784005654.5,
         "_type":"environment"
      }
   ]
}
```

### Actual Behavior

When building a structure of variables it generated a crash that prevented rendering.

### Reproduction Steps

Import data collection JSON attached

### Is there an existing issue for this?

- [X] I have searched the [issue tracker](https://www.github.com/Kong/insomnia/issues) for this problem.

### Additional Information

Version: Insomnia 2022.7.0-beta.6
Build date: 11/29/2022
OS: Darwin x64 22.1.0
Electron: 21.2.0
Node: 16.16.0
V8: 10.6.194.18-electron.0
Architecture: x64

### Insomnia Version

2022.7.0-beta.6

### What operating system are you using?

macOS

### Operating System Version

MacOS Ventura 13

### Installation method

download from https://insomnia.rest/

<img width="1357" alt="insomnia-error-print" src="https://user-images.githubusercontent.com/527641/204688595-5f19c2ec-7fa1-4e02-ad2a-5a2d1c14070c.png">

**Comment by filfreire** [comment]
Hi @524c thanks for reporting this!

I've tried out your collection export, but it looks like the error might be coming from Insomnia trying to parse the GraphQL schema. Any chance you can provide a schema file we can try for us to reproduce, or at least paste what the schema looks like for your `OauthProviders` ?

**Comment by 524c** [comment]
Hi, I recently created a repository to help reproduce a bug in another project.  I can add the resolver that generated this error in Insomnia.  In it you have a graphql server, and so I believe you can reproduce the bug.  What do you think?

**Comment by 524c** [comment]
Hi, @filfreire,
I'm trying to play again, but I can't.
I'm going to retrace my steps from scratch, writing the query again in Insomnia and see if I can understand the whole scenario, then share more details with you. If necessary, I share a repository with a mini project containing a graphql server with resolver, in case it is associated with this bug.

**Comment by 524c** [comment]
Hi @filfreire,

I'm trying to reproduce the bug again, but I can't. I'm going to retrace my steps from scratch, writing the query again in Insomnia and see if I can understand the whole scenario, then share more details with you. If necessary, I share a repository with a mini project containing a graphql server with resolver, in case it is associated with this bug.

**Comment by 0x524c** [comment]
> Hi @524c thanks for reporting this!
> 
> I've tried out your collection export, but it looks like the error might be coming from Insomnia trying to parse the GraphQL schema. Any chance you can provide a schema file we can try for us to reproduce, or at least paste what the schema looks like for your `OauthProviders` ?

https://user-images.githubusercontent.com/527641/204805344-d721434c-4e6f-4554-9f0a-cd1923e56506.mp4

I managed to reproduce. This is a few seconds cut from a video where I retraced my steps. It seems that the problem is when I make a query with a value that does not exist in an enum.

**Comment by 524c** [comment]
Hi @filfreire,


https://user-images.githubusercontent.com/527641/204815245-8579f410-3a4d-4d09-83e9-69e1d75bed01.mp4

The sequence for reproducing the bug and explaining what happens in the video is as follows:
In the terminal I have a graphql server that contains a resolver with an enum of N values.

NOTE: To be able to see the query, I need to do this with the graphql server stopped before open Insomnia or before click on FederatedSignup query.
1) I show what the query is, with the graphql server stopped, otherwise Insominia crashes;
2) start graphql server;
3) in Insomnia, I click on any other query and then on the FederatedSignup query, which makes an invalid query, where the provider field is empty "". The problem also occurs if you pass an invalid value in this field. "google" is invalid for my resolver, and "GOOGLE" is valid.

The problem seems to occur when Insomnia uses the graphql introspection feature.

Before using the repo I provide at the end of the message, you can try with something you already have available, using these things here:

Enum schema
```
enum OauthProviders {
  APPLE
  FACEBOOK
  GOOGLE
}
```

Query
```
mutation FederatedSignup($UserInput: UserInput!, $FUserInput: FUserInput!) {
  FederatedSignup(UserInput: $UserInput, FUserInput: $FUserInput) {
    __typename
    ... on MutationFederatedSignupSuccess {
      data {
        user {
          id
          pk
          provider
          subject
          userId
        }
      }
    }
  }
}
```

Variables
```
{
	"FUserInput": {
		"userId": "108170005730914745289",
		"subject": "user25@gmail.com",
		"provider": ""
	},
	"UserInput": {
		"name": "Rogerio",
		"email": "ser25@gmail.com",
		"password": "123456"
	}
}
```

So, if you can't reproduce the bug with just the above things, you can follow these steps, using this repo with a repro.
To install and run this API, the steps are as follows:

1- clone the repo
`git clone https://github.com/524c/houdini-issue-twoOperations`

then
```
cd houdini-issue-twoOperations/apps/api
git fetch --all
git checkout insomnia-issue
pnpm i
pnpm dev
```

check there
http://127.0.0.1:4000/graphql

NOTE: To be able to see the query, I need to do this with the graphql server stopped before open Insomnia or before click on FederatedSignup query.

**Comment by 0x524c** [comment]
Hi @filfreire,


https://user-images.githubusercontent.com/527641/204815245-8579f410-3a4d-4d09-83e9-69e1d75bed01.mp4

The sequence for reproducing the bug and explaining what happens in the video is as follows:
In the terminal I have a graphql server that contains a resolver with an enum of N values.

NOTE: To be able to see the query, I need to do this with the graphql server stopped before open Insomnia or before click on FederatedSignup query.
1) I show what the query is, with the graphql server stopped, otherwise Insominia crashes;
2) start graphql server;
3) in Insomnia, I click on any other query and then on the FederatedSignup query, which makes an invalid query, where the provider field is empty "". The problem also occurs if you pass an invalid value in this field. "google" is invalid for my resolver, and "GOOGLE" is valid.

The problem seems to occur when Insomnia uses the graphql introspection feature.

Before using the repo I provide at the end of the message, you can try with something you already have available, using these things here:

Enum schema
```prisma
enum OauthProviders {
  APPLE
  FACEBOOK
  GOOGLE
}
```

Query
```graphql
mutation FederatedSignup($UserInput: UserInput!, $FUserInput: FUserInput!) {
  FederatedSignup(UserInput: $UserInput, FUserInput: $FUserInput) {
    __typename
    ... on MutationFederatedSignupSuccess {
      data {
        user {
          id
          pk
          provider
          subject
        }
      }
    }
  }
}
```

Variables
```json
{
	"FUserInput": {
		"userId": "111111111111111111111",
		"subject": "user25@gmail.com",
		"provider": ""
	},
	"UserInput": {
		"name": "Rogerio",
		"email": "ser25@gmail.com",
		"password": "123456"
	}
}
```

So, if you can't reproduce the bug with just the above things, you can follow these steps, using this repo with a repro.
To install and run this API, the steps are as follows:

1- clone the repo
`git clone https://github.com/524c/houdini-issue-twoOperations`

then
```sh
cd houdini-issue-twoOperations/apps/api
git fetch --all
git checkout insomnia-issue
pnpm i
pnpm dev
```

check there
http://127.0.0.1:4000/graphql

NOTE: To be able to see the query, I need to do this with the graphql server stopped before open Insomnia or before click on FederatedSignup query.

**Comment by 524c** [comment]
Hey @filfreire,
I made minor tweaks to the repository I shared. So if you tried to use it before and failed, now everything is ok.

**Comment by RealTirex** [comment]
Is it possible to fix that in near future? Is it a burdensome? Yes

It happens to me, as a result of such steps:
1. I had a valid enum value
2. When I started to change enum value, I got render crash with that error:
![image](https://user-images.githubusercontent.com/26038207/210023732-10ee9182-61dd-4926-9dce-70a2f6096b6d.png)
3. Now, I can't use insomnia
4. It is a common, my friend also got that error.

And stacktrace is here:
```
GraphQLError: Value "P" does not exist in "NetworkPlan" enum.
    at Yl.parseValue (file:///opt/Insomnia/resources/app.asar/index.0869c16c.js:502:10797)
    at Vm (file:///opt/Insomnia/resources/app.asar/index.0869c16c.js:646:20522)
    at Vm (file:///opt/Insomnia/resources/app.asar/index.0869c16c.js:646:19335)
    at file:///opt/Insomnia/resources/app.asar/index.0869c16c.js:646:19879
    at Array.map (<anonymous>)
    at H5 (file:///opt/Insomnia/resources/app.asar/index.0869c16c.js:646:20809)
    at Vm (file:///opt/Insomnia/resources/app.asar/index.0869c16c.js:646:19659)
    at Vm (file:///opt/Insomnia/resources/app.asar/index.0869c16c.js:646:19335)
    at file:///opt/Insomnia/resources/app.asar/index.0869c16c.js:646:19019
    at Array.forEach (<anonymous>)
    at UCe (file:///opt/Insomnia/resources/app.asar/index.0869c16c.js:646:18928)
    at file:///opt/Insomnia/resources/app.asar/index.0869c16c.js:646:18866
    at x (file:///opt/Insomnia/resources/app.asar/index.0869c16c.js:431:14293)
    at ii.<anonymous> (file:///opt/Insomnia/resources/app.asar/index.0869c16c.js:431:16262)
    at file:///opt/Insomnia/resources/app.asar/index.0869c16c.js:407:10355
    at ii.setOption (file:///opt/Insomnia/resources/app.asar/index.0869c16c.js:412:1978)
    at file:///opt/Insomnia/resources/app.asar/index.0869c16c.js:671:5056
    at mE (file:///opt/Insomnia/resources/app.asar/index.0869c16c.js:67:24458)
    at nh (file:///opt/Insomnia/resources/app.asar/index.0869c16c.js:67:42799)
    at file:///opt/Insomnia/resources/app.asar/index.0869c16c.js:67:41058
    at A (file:///opt/Insomnia/resources/app.asar/index.0869c16c.js:52:1648)
    at Immediate.ce [as _onImmediate] (file:///opt/Insomnia/resources/app.asar/index.0869c16c.js:52:2022)
    at process.processImmediate (node:internal/timers:466:21)
```

**Comment by filfreire** [comment]
Duplicate of #5625

**Comment by filfreire** [comment]
Hey folks, I closed this one by mistake and meant to close #5625 instead as duplicate of this one 😅 

In any case, this will be fixed in an upcoming patch release after we merge https://github.com/Kong/insomnia/pull/5650

**Comment by 0x524c** [comment]
> Hey folks, I closed this one by mistake and meant to close #5625 instead as duplicate of this one 😅
> 
> In any case, this will be fixed in an upcoming patch release after we merge #5650

Okay, I figured that was it.
I will test the new version. Thanks 👍
